pandas/core/tools/datetimes.py

```diff
@@ -514,11 +464,17 @@ def _array_strptime_with_fallback(
     """
     Call array_strptime, with fallback behavior depending on 'errors'.
     """
-    result, timezones = array_strptime(arg, fmt, exact=exact, errors=errors, utc=utc)
-    if any(tz is not None for tz in timezones):
-        return _return_parsed_timezone_results(result, timezones, utc, name)
-
-    return _box_as_indexlike(result, utc=utc, name=name)
+    result, tz_out = array_strptime(arg, fmt, exact=exact, errors=errors, utc=utc)
+    if tz_out is not None:
+        dtype = DatetimeTZDtype(tz=tz_out)
+        dta = DatetimeArray._simple_new(result, dtype=dtype)
+        if utc:
+            dta = dta.tz_convert("UTC")
+        return Index(dta, name=name)
+    elif result.dtype != object and utc:
+        res = Index(result, dtype="M8[ns, UTC]", name=name)
```

Comment:
Are there cases where this would not be `ns`?

Reply:
not until 3.0 when inference resolution goes live